rest/src/test/java/com/ericsson/bss/cassandra/ecchronos/rest/TestRepairSchedulerRESTImpl.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright 2019 Telefonaktiebolaget LM Ericsson
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.ericsson.bss.cassandra.ecchronos.rest;
+
+import com.datastax.driver.core.Host;
+import com.ericsson.bss.cassandra.ecchronos.core.repair.RepairJobView;
+import com.ericsson.bss.cassandra.ecchronos.core.repair.RepairScheduler;
+import com.ericsson.bss.cassandra.ecchronos.core.utils.LongTokenRange;
+import com.ericsson.bss.cassandra.ecchronos.rest.types.CompleteRepairJob;
+import com.ericsson.bss.cassandra.ecchronos.rest.types.ScheduledRepairJob;
+import com.ericsson.bss.cassandra.ecchronos.rest.types.VnodeState;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import org.assertj.core.util.Sets;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class TestRepairSchedulerRESTImpl
+{
+    @Mock
+    private RepairScheduler myRepairScheduler;
+
+    @Test
+    public void testListEmpty()
+    {
+        when(myRepairScheduler.getCurrentRepairJobs()).thenReturn(new ArrayList<>());
+
+        RepairSchedulerREST repairSchedulerService = new RepairSchedulerRESTImpl(myRepairScheduler);
+
+        List<ScheduledRepairJob> response = new Gson().fromJson(repairSchedulerService.list(), new TypeToken<List<ScheduledRepairJob>>(){}.getType());
```

Comment:
Maybe the Type (new TypeToken<List<ScheduledRepairJob>>(){}.getType()) could be broken out into a constant?